journalbeat/reader/journal.go

```diff
@@ -303,32 +260,16 @@ func (r *Reader) toEvent(entry *sdjournal.JournalEntry) *beat.Event {
 	return &event
 }
 
-func isKept(key string) bool {
-	return key != ""
-}
-
-// stopOrWait waits for a journal event.
-func (r *Reader) stopOrWait() {
-	select {
-	case <-r.done:
-	case r.changes <- r.journal.Wait(100 * time.Millisecond):
-	}
-}
-
-func (r *Reader) wait() {
-	select {
-	case <-r.done:
-		return
-	case <-time.After(r.config.Backoff):
-	}
-
-	if r.config.Backoff < r.config.MaxBackoff {
-		r.config.Backoff = r.config.Backoff * time.Duration(r.config.BackoffFactor)
-		if r.config.Backoff > r.config.MaxBackoff {
-			r.config.Backoff = r.config.MaxBackoff
+func (r *Reader) convertNamedField(fc fieldConversion, value string) interface{} {
+	if fc.isInteger {
+		v, err := strconv.ParseInt(value, 10, 64)
+		if err != nil {
+			r.logger.Debugf("Failed to convert field: %s \"%v\" to int: %v", fc.name, value, err)
+			return value
```

Comment:
Will the event still be indexible if this one fails?

Will this clash with fields.yml and the index mapping?

Reply:
I doubt that this check would fail. I put it here as a "sentinel". Only those fields are attempted to be converted to integer which are part of `journalbeat/reader/fields.go` where only journald or docker fields are present. The type of these fields are known in advance, because I tested it and read the documentation.

There is also documentation on each field of systemd journal here: https://www.freedesktop.org/software/systemd/man/systemd.journal-fields.html
Docker journald fields: https://docs.docker.com/config/containers/logging/journald/

Unknown fields are stored under `custom` key in the event. This is not indexed, because its structure is unknown in advance. I thought about indexing it as a nested object in ES. WDYT?

Reply:
I added `custom` to `fields.yml` to be indexed as a nested object. If you don't agree with the change I will drop the commit.